ansible-test units fails during pytest configuration parsing on python:3.6-buster docker image

##### SUMMARY
Running `ansible-test units` command in the python:3.6-buster docker fails when pytest tries to parse its configuration file.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ansible-test

##### ANSIBLE VERSION
```paste below
ansible 2.10.1rc3
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/tadej/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/tadej/ansible_collections/sensu/venv/lib64/python3.8/site-packages/ansible
  executable location = /home/tadej/ansible_collections/sensu/venv/bin/ansible
  python version = 3.8.5 (default, Aug 12 2020, 00:00:00) [GCC 10.2.1 20200723 (Red Hat 10.2.1-1)]
```

##### CONFIGURATION
<!--- Paste verbatim output from "ansible-config dump --only-changed" between quotes -->
```paste below

```

##### OS / ENVIRONMENT
python:3.6-buster docker image running on podman (tested on Fedora 32) or docker (tested on CircleCI).


##### STEPS TO REPRODUCE
1. Start a container from python:3.6-buster image.
2. Install ansible-base >= 2.10 or ansible >= 2.10
3. Clone any Ansible collection that has at least one unit test.
4. Run `ansible-test units --python 3.6 --requirements`.

##### EXPECTED RESULTS
Ansible should run the tests.


##### ACTUAL RESULTS
Pytest fails to parse the configuration file ansible-test provides. Details below:

<!--- Paste verbatim command output between quotes -->
```paste below
$ ansible-test units --python 3.6 --coverage --requirements
Collecting coverage<5.0.0,>=4.5.1
  Downloading coverage-4.5.4-cp36-cp36m-manylinux1_x86_64.whl (205 kB)
     |################################| 205 kB 10.1 MB/s eta 0:00:01
Collecting mock>=2.0.0
  Downloading mock-4.0.2-py3-none-any.whl (28 kB)
Collecting pytest-mock>=1.4.0
  Downloading pytest_mock-3.3.1-py3-none-any.whl (11 kB)
Collecting pytest
  Downloading pytest-6.0.2-py3-none-any.whl (270 kB)
     |################################| 270 kB 34.8 MB/s eta 0:00:01
Collecting pytest-xdist
  Downloading pytest_xdist-2.1.0-py3-none-any.whl (36 kB)
Collecting pluggy<1.0,>=0.12
  Downloading pluggy-0.13.1-py2.py3-none-any.whl (18 kB)
Collecting iniconfig
  Downloading iniconfig-1.0.1-py3-none-any.whl (4.2 kB)
Collecting py>=1.8.2
  Downloading py-1.9.0-py2.py3-none-any.whl (99 kB)
     |################################| 99 kB 19.3 MB/s eta 0:00:01
Collecting attrs>=17.4.0
  Downloading attrs-20.2.0-py2.py3-none-any.whl (48 kB)
     |################################| 48 kB 12.1 MB/s eta 0:00:01
Collecting importlib-metadata>=0.12; python_version < "3.8"
  Downloading importlib_metadata-1.7.0-py2.py3-none-any.whl (31 kB)
Collecting toml
  Downloading toml-0.10.1-py2.py3-none-any.whl (19 kB)
Collecting more-itertools>=4.0.0
  Downloading more_itertools-8.5.0-py3-none-any.whl (44 kB)
     |################################| 44 kB 7.0 MB/s  eta 0:00:01
Collecting pytest-forked>=1.0.2
  Downloading pytest_forked-1.3.0-py2.py3-none-any.whl (4.7 kB)
Collecting execnet>=1.1
  Downloading execnet-1.7.1-py2.py3-none-any.whl (39 kB)
Collecting zipp>=0.5
  Downloading zipp-3.1.0-py3-none-any.whl (4.9 kB)
Collecting apipkg>=1.4
  Downloading apipkg-1.5-py2.py3-none-any.whl (4.9 kB)
Installing collected packages: coverage, py, zipp, importlib-metadata, pluggy, iniconfig, attrs, toml, more-itertools, pytest, pytest-forked, mock, pytest-mock, apipkg, execnet, pytest-xdist
Successfully installed apipkg-1.5 attrs-20.2.0 coverage-4.5.4 execnet-1.7.1 importlib-metadata-1.7.0 iniconfig-1.0.1 mock-4.0.2 more-itertools-8.5.0 pluggy-0.13.1 py-1.9.0 pytest-6.0.2 pytest-forked-1.3.0 pytest-mock-3.3.1 pytest-xdist-2.1.0 toml-0.10.1 zipp-3.1.0
Unit test with Python 3.6
Traceback (most recent call last):
  File "/home/circleci/venv/lib/python3.6/site-packages/pytest/__main__.py", line 7, in <module>
    raise SystemExit(pytest.console_main())
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/__init__.py", line 180, in console_main
    code = main()
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/__init__.py", line 136, in main
    config = _prepareconfig(args, plugins)
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/__init__.py", line 314, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/home/circleci/venv/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/circleci/venv/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/circleci/venv/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/circleci/venv/lib/python3.6/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/helpconfig.py", line 99, in pytest_cmdline_parse
    config = outcome.get_result()  # type: Config
  File "/home/circleci/venv/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/circleci/venv/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/__init__.py", line 932, in pytest_cmdline_parse
    self.parse(args)
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/__init__.py", line 1204, in parse
    self._preparse(args, addopts=addopts)
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/__init__.py", line 1085, in _preparse
    self._initini(args)
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/__init__.py", line 1009, in _initini
    config=self,
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/findpaths.py", line 170, in determine_setup
    inicfg = load_config_dict_from_file(inipath_) or {}
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/findpaths.py", line 42, in load_config_dict_from_file
    iniconfig = _parse_ini_config(filepath)
  File "/home/circleci/venv/lib/python3.6/site-packages/_pytest/config/findpaths.py", line 27, in _parse_ini_config
    return iniconfig.IniConfig(path)
  File "/home/circleci/venv/lib/python3.6/site-packages/iniconfig.py", line 54, in __init__
    tokens = self._parse(iter(f))
  File "/home/circleci/venv/lib/python3.6/site-packages/iniconfig.py", line 82, in _parse
    for lineno, line in enumerate(line_iter):
  File "/usr/local/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 230: ordinal not in range(128)
ERROR: Command "pytest --boxed -r a -n auto --color yes -p no:cacheprovider -c /home/circleci/venv/lib/python3.6/site-packages/ansible_test/_data/pytest.ini --junit-xml /home/circleci/ansible_collections/sensu/sensu_go/tests/output/junit/python3.6-units.xml --strict-markers tests/unit/action/test_bonsai_asset.py" returned exit status 1.
make: *** [Makefile:39: units] Error 1
```

##### OTHER INFO

Digging around a bit, we managed to discover that there are two factors that contribute to the failure.

 1. `test/lib/ansible_test/_data/pytest.ini` file contains an em-dash.
 2. Python in python:3.6-buster image switches its preferred encoding to ASCII when the *LC_ALL* environment variable is set to `en_US.UTF-8`.

```
$ podman run --rm -e LC_ALL=C.UTF-8 python:3.6-buster python -c 'import locale; print(locale.getpreferredencoding())'
UTF-8
$ podman run --rm -e LC_ALL=en_US.UTF-8 python:3.6-buster python -c 'import locale; print(locale.getpreferredencoding())'
ANSI_X3.4-1968
```

If we remove the non-ASCII dash from the configuration file, things start working as expected.

--- a/changelogs/fragments/71739-remove-em-dash-from-pytest-config.yml
+++ b/changelogs/fragments/71739-remove-em-dash-from-pytest-config.yml
@@ -1,0 +1,4 @@
+minor_changes:
+- ansible-test - Remove em dash from the Pytest configuration file in order to
+  be readable on systems where preferred encoding is ASCII.
+  (https://github.com/ansible/ansible/issues/71739)

--- a/test/lib/ansible_test/_data/pytest.ini
+++ b/test/lib/ansible_test/_data/pytest.ini
@@ -3,7 +3,7 @@ xfail_strict = true
 mock_use_standalone_module = true
 # It was decided to stick with "legacy" (aka "xunit1") for now.
 # Currently used pytest versions all support xunit2 format too.
-# Except the one used under Python 2.6 — it doesn't process this option
+# Except the one used under Python 2.6 - it doesn't process this option
 # at all. Ref:
 # https://github.com/ansible/ansible/pull/66445#discussion_r372530176
 junit_family = xunit1

